src/Swarm/Game/Scenario/Status.hs

```diff
@@ -0,0 +1,235 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Swarm.Game.Scenario.Status where
```

Comment:
If I wrote modules `Scoring.X`, `Scoring.Y` and `Scoring.Z`, I might also write module `Scoring` to reexport the functions which are not internal to the implementation.

This looks like that module. Why is it named `Status` when it deals a lot with scores? :raised_eyebrow: 

If I had to guess what would be in a `Status` module, I would consider the `Objective` state tracking. But I do not see that here. This makes me confused.

Reply:
**Why is `ScenarioInfo` not in the `ScenarioInfo` module?!**

@kostmo please treat modules like a helpful tool, not an annoying constraint to bend and break.

Reply:
> please treat modules like a helpful tool, not an annoying constraint to bend and break.

@xsebek I can understand you're frustrated, but I'm not sure how helpful this kind of comment is (attributing certain motivations or attitudes to others).  Let's try to stick to making concrete suggestions on how to improve.

Reply:
Also, you forgot the lenses in the `ScenarioInfo` module:
```Haskell
makePrisms ''ScenarioItem
makePrisms ''ScenarioStatus
```
They might not be used much, so you might be able to delete them.

Honestly, I am surprised TH for generating instances even works in a different module, I have never seen it done that way.

Reply:
Appreciate the reminder about appropriate communication, @byorgey.

I'll make another pass at module organization/docs.

Reply:
> lenses in the `ScenarioInfo` module

Tried moving them but got some odd "Not in scope" errors.  Can follow up on this in a later PR.